docs: claim covenant use cases, misc fixes (#1118)
BoltzExchange/boltz-backend: docs/claim-covenants.md

approach maintains minimal trust assumptions.

**This is the recommended way to use claim covenants.**

### LNURL/LN-Address Spontaneous Offline Payments

When using claim covenants for **spontaneous payments via LNURL or Lightning
addresses** while the swap client is offline, they **cannot** be used in a
trust-minimized manner:

In this scenario, the sender pulls payment details on-demand (e.g., via LNURL or
a Lightning address), but the receiving client is offline and cannot validate
the covenant script setup. The swap creator retains sole control over the actual
claiming conditions, and the offline recipient has no way to verify correctness.
From a trust perspective, this is similar to providing an xpub or wallet
descriptor as the swap destination directly to the swap creator.

**We do NOT recommend using claim covenants this way.**

### General Considerations

When handing over the preimage of a Reverse Swap to the swap creator, which is
usually a third party like a mobile wallet provider, you have to rely on this
party not to collude with the Lightning node that accepts HTLCs for the hold
invoice. If that happens, the covenant script path spend would not be executed,

> but the Lightning HTLCs would be resolved and eventually, the coins locked on

Reviewer: ```suggestion
while the Lightning HTLCs are resolved and eventually, the coins locked on
```
